### PR TITLE
Bloggerly: Creating integration tests and fixing n+1 Problem

### PR DESCRIPTION
**Pull request by hernandanielzamora**
# In this PR, the following changes were made:

## Fixing the n+1 Problem

Fixed the n+1 query problem using `includes`

Before code refactor:

After code refactor:

## Creating integration tests with Capybara

Tests created: 

- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.

- User show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

### CODE REVIEW

File: spec/system/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe 'Post Index', type: :system do
+  before do
```
Comment (Lizdev-05) [inline spec/system/post_index_spec.rb:4]: - [x] Please you are missing a test that tests you can see a section for pagination if there are more posts than fit on the view  :pray:

File: spec/system/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+RSpec.describe 'User Show Page', type: :system do
+  before do
```
Comment (Lizdev-05) [inline spec/system/user_show_spec.rb:4]: - [x] Please you a missing a test that tests that a user redirects to the post's show page when you click on a user's post. :blush:

**Review by Lizdev-05 — CHANGES_REQUESTED** [review]
Hi @hernandanielzamora and @zamcham 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
:heavy_check_mark: Professional Readme file :+1: 
:heavy_check_mark: Good commit history :+1:
:heavy_check_mark: No linter error :+1: 
:heavy_check_mark: N+1 problem fixed :+1: 

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please make sure that the Capybara gem is in the `Gemfile` file for the integration test :pray: 
  
  ```ruby
   group :development, :test do
     ...
     **gem 'capybara'**
     ...
   end
  ```

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi Team,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  No Linters errors 🔥
- [ ]  Great job working on the tests💯
- [ ]  Followed GitFlow ⭐🚀
- [ ]  Descrtipive PR⭐
- [ ] N+1 problem is solved well 🚀🙏🏻


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**